processors/src/RefittedTracksProcessor.cxx

```diff
@@ -59,18 +61,56 @@ bool RefittedTracksProcessor::process(IEvent* ievent) {
   //Get all the rawHits fits
   EVENT::LCCollection* raw_svt_hit_fits = event->getLCCollection(Collections::RAW_SVT_HIT_FITS);
 
-  //Initialize map of shared hits
+  //Grab the vertices and the vtx candidates
+  EVENT::LCCollection* u_vtx_candidates = nullptr;
+  EVENT::LCCollection* u_vtxs = nullptr;
+  if (utils::hasCollection(event->getLCEvent(),Collections::UC_V0CANDIDATES)) { 
+    //Get the vertex candidates
+    u_vtx_candidates  = event->getLCCollection(Collections::UC_V0CANDIDATES);
+    //Get the vertices 
+    u_vtxs = event->getLCCollection(Collections::UC_V0VERTICES);
+  
+    
+    for (int ivtx = 0 ; ivtx < u_vtxs->getNumberOfElements(); ++ivtx) {
+      Vertex* vtx = utils::buildVertex(static_cast<EVENT::Vertex*>(u_vtxs->getElementAt(ivtx)));
+      vertices_.push_back(vtx);
```

Comment:
IF buildVertex creates std::unique_ptr, replace with vertices_.push_back(std::move(vtx))

IF not going with an std::move(), consider these two lines to be replace with vertices_.emplace_back() and instead of buildVertex, have Vertex object created with a proper constructor.

Reply:
Hi Maurik, 

In principle I'd like to have a set of utilities classes that take care of converting LCIO objects to ROOT objects that can be used by different processors. 
Basically I'd like to have the object conversion and ntuple building done in separate code. That's why I don't want to have constructors in the processor but internal to utilities.

Reply:
Hi PF,
Thank you for the answers!
I noticed that the EcalDataProcessor and SVTDataProcessor for instance, store the objects in TClonesArray* while the TrackingProcessor and RefittedTrackingProcessor store objects in vector<>.  I don't actually know for sure which is "better" in terms of performance or general usability, but it seems odd to have two different ways of doing this. Once we have a lot of root files written, this will be difficult to change, so perhaps we should think a bit more about the pros and cons of each and then decide what is best. Just saw the other pull request, where you state that you prefer vector<>. Is that for usability or for performance reasons?

As for utilities, vs everything in a processor, vs a constructor on the object (that is part of the event library, not processors), I would prefer the latter, unless there is a reason to keep all LCIO out of "event". That last reason is actually solid, but in that case, we should consider fixing the cmake files to make is possible to compile the "event" library without having LCIO install at all. I.e. if LCIO is not on your system, hpstr only compiles the libevent.so. That way the root files can be read and analyzed without needing LCIO, simplifying use.